nvflare/tool/code_pre_installer/install.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import shutil
+import tempfile
+from pathlib import Path
+from zipfile import ZipFile
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="pre-Install application code and libs")
+    parser.add_argument("--app-code", required=True, help="Path to application code zip file")
+    parser.add_argument(
+        "--install-prefix", default="/opt/nvflare/apps", help="Installation prefix (default: /opt/nvflare/apps)"
+    )
+    parser.add_argument("--site-name", required=True, help="Target site name (e.g., site-1, server)")
+    return parser.parse_args()
+
+
+def install_requirements(requirements_file: Path):
+    """Install Python packages from requirements.txt."""
+    if not requirements_file.exists():
+        print("No requirements.txt found, skipping package installation")
+        return
+
+    print(f"Installing packages from {requirements_file}...")
+    import subprocess
+
+    try:
+        subprocess.run(["pip", "install", "-r", str(requirements_file)], check=True)
+    except subprocess.CalledProcessError as e:
+        raise ValueError(f"Failed to install requirements: {e}")
+
+
+def install_app_code(app_code: Path, install_prefix: Path, site_name: str):
+    """Install NVFLARE application code for a specific site."""
+    CUSTOM_DIR = Path("/local/custom")
```

Comment:
we might want to make this a constant so it is easier to maintain. could be done in future though